shiny/ui/_theme_brand.py

```diff
@@ -0,0 +1,410 @@
+                            )
+
+                        sass_vars_typography[typo_sass_var] = typ_field_value
+                    elif raise_for_unmapped_vars:
+                        raise ThemeBrandUnmappedFieldError(
+                            f"typography.{typ_field}.{typ_field_key}"
+                        )
+
+        # Theme -----------------------------------------------------------------------
+        sass_vars_brand: dict[str, str] = {
+            **sass_vars_brand_colors,
+            **sass_vars_colors,
+            **sass_vars_typography,
+        }
+        sass_vars_brand = {k: v for k, v in sass_vars_brand.items()}
+
+        # Defaults are added in reverse order, so each chunk appears above the next
+        # layer of defaults. The intended order in the final output is:
+        # 1. Brand Sass vars (colors, typography)
+        # 2. Brand Bootstrap Sass vars
+        # 3. Fallback vars needed by additional Brand rules
+        self.add_defaults(
+            # Variables we create to augment Bootstrap's variables
+            **{
+                "code-font-weight": None,
+                "code-inline-font-weight": None,
+                "code-inline-font-size": None,
+                "code-block-font-weight": None,
+                "code-block-font-size": None,
+                "link-bg": None,
+                "link-weight": None,
+                "gray-100": "mix($white, $black, 90%)",
+                "gray-200": "mix($white, $black, 80%)",
+                "gray-300": "mix($white, $black, 70%)",
+                "gray-400": "mix($white, $black, 60%)",
+                "gray-500": "mix($white, $black, 50%)",
+                "gray-600": "mix($white, $black, 40%)",
+                "gray-700": "mix($white, $black, 30%)",
+                "gray-800": "mix($white, $black, 20%)",
+                "gray-900": "mix($white, $black, 10%)",
+            }
+        )
+        self.add_defaults(**brand_bootstrap.defaults)
+        self.add_defaults(**sass_vars_brand)
+        # Brand Rules ----
+        self.add_rules(":root {", *css_vars_brand_colors, "}")
+        # Additional rules to fill in Bootstrap styles for Brand parameters
+        self.add_rules(
+            """
+            // https://github.com/twbs/bootstrap/blob/5c2f2e7e/scss/_root.scss#L82
+            :root {
+                --#{$prefix}link-bg: #{$link-bg};
+                --#{$prefix}link-weight: #{$link-weight};
+            }
+            // https://github.com/twbs/bootstrap/blob/5c2f2e7e/scss/_reboot.scss#L244
+            a {
+                background-color: var(--#{$prefix}link-bg);
+                font-weight: var(--#{$prefix}link-weight);
+            }
+            code {
+                font-weight: $code-font-weight;
+            }
+            code {
+              font-weight: $code-inline-font-weight;
+              font-size: $code-inline-font-size;
+            }
+            pre {
```

Comment:
```suggestion
            pre code {
```

Reply:
No, this follows bootstraps lead here

Reply:
https://github.com/twbs/bootstrap/blob/30e01525670386c23f9bc0d9752a3f8f70af73cc/scss/_reboot.scss#L287